### ISSUE TITLE
Raspberry Pi Stretch MAKE Error

### ISSUE THREAD

**Issue by PeterESP8266** [vysheng/tg#1535]
I'm trying to install Telegram on a Raspberry Pi running the latest stretch version.
Following the installation guide I end up with the following errors after the MAKE command:

`gcc -I. -I. -I./tgl -g -O2  -I/usr/local/include -I/usr/include -I/usr/include -I/usr/include/lua5.2  -DHAVE_CONFIG_H -Wall -Werror -Wextra -Wno-missing-field-initializers -Wno-deprecated-declarations -fno-strict-aliasing -fno-omit-frame-pointer -ggdb -Wno-unused-parameter -fPIC -iquote ./tgl -c -MP -MD -MF dep/crypto/rsa_pem_openssl.d -MQ objs/crypto/rsa_pem_openssl.o -o objs/crypto/rsa_pem_openssl.o tgl/crypto/rsa_pem_openssl.c
tgl/crypto/rsa_pem_openssl.c: In function 'TGLC_rsa_new':
tgl/crypto/rsa_pem_openssl.c:41:6: error: dereferencing pointer to incomplete type 'RSA {aka struct rsa_st}'
   ret->e = unwrap_bn (TGLC_bn_new ());
      ^~
tgl/crypto/rsa_pem_openssl.c: In function 'TGLC_rsa_n':
tgl/crypto/rsa_pem_openssl.c:52:1: error: control reaches end of non-void function [-Werror=return-type]
 RSA_GETTER(n);
 ^~~~~~~~~~
tgl/crypto/rsa_pem_openssl.c: In function 'TGLC_rsa_e':
tgl/crypto/rsa_pem_openssl.c:53:1: error: control reaches end of non-void function [-Werror=return-type]
 RSA_GETTER(e);
 ^~~~~~~~~~
cc1: all warnings being treated as errors
Makefile.tgl:20: recipe for target 'objs/crypto/rsa_pem_openssl.o' failed
make: *** [objs/crypto/rsa_pem_openssl.o] Error 1`

after that, when trying to run ` ./bin/telegram-cli -k tg-server.pub`
i get the error message:
`-bash: ./bin/telegram-cli: No such file or directory`

**Comment by hbh7** [comment]
Got the same error when compiling. Seems the automated build fails too. I guess they just need to update the repo to make the code actually compile.... ¯\_(ツ)_/¯

**Comment by PeterESP8266** [comment]
Thanks for the help. Could you tell me how you managed to install that certain libssl1.0-dev package?

**Comment by JugglerMaster** [comment]
The libssl1.0 package is what fixed this for me. The ./configure command does not let you know you need it.

**Comment by hbh7** [comment]
Yes sorry for not responding; I thought I had already but apparently not. Anyways, if you're on Ubuntu like me, just do `sudo apt-get install libssl1.0-dev`
